[stable/vmware_exporter] Introduce a new helm chart for vmware_exporter

kremers opened the pull request:
==> Linting .
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, no failures

Signed-off-by: Martin Kremers <info@martinkremers.de>

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

kremers committed:
[stable/vmware_exporter] Introduce a new helm chart for vmware_exporter

Signed-off-by: Martin Kremers <info@martinkremers.de>

mtparet commented on stable/vmware-exporter/OWNERS:
```diff
@@ -0,0 +1,4 @@
+approvers:
+- kremers
```
You need at least two approvers/reviewers. (got the same comment on adding a new chart)

kremers replied:
Thx! Added sameersbn as suggested in the PR here

mtparet commented on stable/vmware-exporter/templates/secret.yml:
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
```
It would be a good practive allowing using an existing secret.

kremers replied:
Yes :) 
I would like to add that additionally in another PR if possible.

kremers committed:
[stable/vmware_exporter] Health-Check on index page

If the collection of metrics takes very long, it makes sense not
to query the /metrics endpoint for healthchecks.
This can point to / or /healthz

- Prevents intense load on VCenter API

Signed-off-by: Martin Kremers <info@martinkremers.de>

mtparet commented on stable/vmware-exporter/templates/deployment.yaml:
```diff
@@ -0,0 +1,56 @@
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 80
```
We should expose 9272 directly, no ?

mtparet commented on stable/vmware-exporter/values.yaml:
```diff
@@ -0,0 +1,75 @@
+  tag: v0.7.4
+  pullPolicy: IfNotPresent
+
+service:
```
Do we need to enable service ? The scrapping from prometheus is done at the pod level.

kremers replied:
Being able to retrieve metrics via ingress would be great for testing and debugging purposes. I can switch this off in the defaults. Any thougts?

chrisz100 replied:
Defaults (as insecure as they are) tend to stay. I'd disable

kremers replied:
Fixed with service.enabled: false as default

kremers committed:
[charts/vmware_exporter] Extended readme with sharding example

- Minor issues fixed regarding the service/ingress port
- Put a sharding example in place, see README.md

Signed-off-by: Martin Kremers <info@martinkremers.de>

kremers committed:
[chart/vmware_exporter] Approvers and maintainer section reworked

Signed-off-by: Martin Kremers <info@martinkremers.de>

chrisz100 commented on stable/vmware-exporter/templates/NOTES.txt:
```diff
@@ -0,0 +1,19 @@
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "vmware-exporter.name" . }},release={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}")
+  echo "Visit http://127.0.0.1:8080 to use your application"
+  kubectl port-forward $POD_NAME 8080:80
+{{- end }}
```
Is all of this required? node-exporter is not really something you'd open up in a browser so that template looks a little misleading.

kremers replied:
Added a better NOTES.txt in experimental/vmware_exporter claiming how to scrape with Prometheus kubernetes_sd.

Need still to test the output under different circumstances as mentioned in the README.md

chrisz100 commented on stable/vmware-exporter/OWNERS:
```diff
@@ -0,0 +1,10 @@
+- kremers
+- pryorda
+- mtparet
+- chrisz100
```
I don't know about the helm project but as reviewer permissions are included in approvers the kubernetes best practice is not include people in both approvers AND reviewers. Just put them into approvers as this supersedes.

kremers committed:
[chart/vmware_exporter] Disable service by default

Signed-off-by: Martin Kremers <info@martinkremers.de>

kremers committed:
[chart/vmware_exporter] Use service port in ingress

Signed-off-by: Martin Kremers <info@martinkremers.de>

kremers committed:
[chart/vmware_exporter] Added improved NOTES.txt

Signed-off-by: Martin Kremers <info@martinkremers.de>
